apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -133,6 +141,40 @@ export function TokenDetails({ coinType }: TokenDetailsProps) {
             !!supplyIncreaseVestingObjectsStaked?.pages?.[0]?.data?.length;
     }
 
+    if (migrationEnabled) {
+        const { data: basicOutputObjects } = useGetOwnedObjects(
+            activeAccountAddress || '',
+            { StructType: STARDUST_BASIC_OUTPUT_TYPE },
+            OBJECT_PER_REQ,
+        );
+
+        const { data: nftOutputObjects } = useGetOwnedObjects(
+            activeAccountAddress || '',
+            { StructType: STARDUST_NFT_OUTPUT_TYPE },
+            OBJECT_PER_REQ,
+        );
+
+        needsMigration =
+            !!basicOutputObjects?.pages?.[0]?.data?.length ||
+            !!nftOutputObjects?.pages?.[0]?.data?.length;
+
+        if (!needsMigration) {
+            const { stardustIndexerClient } = useStardustIndexerClientContext();
+
+            const indexedBasicOutputs = stardustIndexerClient?.getBasicOutputs(
```

Comment:
instead of this approach we could probably do something like https://github.com/iotaledger/iota/blob/3e3cd6a692689f49581cef614ab64746ea3073e8/apps/wallet-dashboard/hooks/useGetAllStardustSharedObjects.ts#L18 to use react query, we could reuse that hook and move it to core

Reply:
also because the responses now are promises so needsMigration is always true 🙏🏼